apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/Version2CustomTypeAdapterToAdapter.kt

```diff
@@ -0,0 +1,73 @@
+package com.apollographql.apollo3.api
+
+import com.apollographql.apollo3.api.json.JsonReader
+import com.apollographql.apollo3.api.json.JsonWriter
+import kotlin.jvm.JvmStatic
+
+/**
+ * An [Adapter] that wraps an Apollo Android v2 style [CustomTypeAdapter], to ease migration from v2 to v3.
+ */
+@Deprecated("Used for backward compatibility with 2.x, use Adapter instead")
+class Version2CustomTypeAdapterToAdapter<T>(
+    private val v2CustomTypeAdapter: CustomTypeAdapter<T>,
+) : Adapter<T> {
+  override fun fromJson(reader: JsonReader, customScalarAdapters: CustomScalarAdapters): T {
+    val value: Any? = if (reader.peek() == JsonReader.Token.NULL) null else AnyAdapter.fromJson(reader)
```

Comment:
Nit: there is a `NullableAnyAdapter` : https://github.com/apollographql/apollo-android/blob/d852dd92ee084732d49ae993f4e1346384f2fde7/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/DefaultAdapters.kt#L224

Reply:
Neat! I'll amend.